connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/RemoteSparkSession.scala

```diff
@@ -0,0 +1,241 @@
+      // scalastyle:off println
+      println(
+        "Will start Spark Connect server with `spark.sql.catalogImplementation=in-memory`, " +
+          "some tests that rely on Hive will be ignored. If you don't want to skip them:\n" +
+          "1. Test with maven: run `build/mvn install -DskipTests -Phive` before testing\n" +
+          "2. Test with sbt: run test with `-Phive` profile")
+      // scalastyle:on println
+      // SPARK-43647: Proactively cleaning the `classes` and `test-classes` dir of hive
+      // module to avoid unexpected loading of `DataSourceRegister` in hive module during
+      // testing without `-Phive` profile.
+      IntegrationTestUtils.cleanUpHiveClassesDirIfNeeded()
+      "in-memory"
+    }
+    val confs = Seq(
+      // Use InMemoryTableCatalog for V2 writer tests
+      "spark.sql.catalog.testcat=org.apache.spark.sql.connector.catalog.InMemoryTableCatalog",
+      // Try to use the hive catalog, fallback to in-memory if it is not there.
+      "spark.sql.catalogImplementation=" + catalogImplementation,
+      // Make the server terminate reattachable streams every 1 second and 123 bytes,
+      // to make the tests exercise reattach.
+      "spark.connect.execute.reattachable.senderMaxStreamDuration=1s",
+      "spark.connect.execute.reattachable.senderMaxStreamSize=123",
+      // Disable UI
+      "spark.ui.enabled=false")
+    Seq("--jars", catalystTestJar) ++ confs.flatMap(v => "--conf" :: v :: Nil)
+  }
+
+  def start(): Unit = {
+    assert(!stopped)
+    sparkConnect
+  }
+
+  def stop(): Int = {
+    stopped = true
+    debug("Stopping the Spark Connect Server...")
+    try {
+      consoleOut.write(serverStopCommand.getBytes)
+      consoleOut.flush()
+      consoleOut.close()
+      if (!sparkConnect.waitFor(2, TimeUnit.SECONDS)) {
+        sparkConnect.destroyForcibly()
+      }
+      val code = sparkConnect.exitValue()
+      debug(s"Spark Connect Server is stopped with exit code: $code")
+      code
+    } catch {
+      case e: IOException if e.getMessage.contains("Stream closed") =>
+        -1
+      case e: Throwable =>
+        debug(e)
+        sparkConnect.destroyForcibly()
+        throw e
+    }
+  }
+
+  def syncTestDependencies(spark: SparkSession): Unit = {
+    // Both SBT & Maven pass the test-classes as a directory instead of a jar.
+    val testClassesPath = Paths.get(IntegrationTestUtils.connectClientTestClassDir)
+    spark.client.artifactManager.addClassDir(testClassesPath)
+
+    // We need scalatest & scalactic on the classpath to make the tests work.
+    val jars = System
+      .getProperty("java.class.path")
+      .split(File.pathSeparatorChar)
+      .filter(_.endsWith(".jar"))
+      .filter(e => e.contains("scalatest") || e.contains("scalactic"))
+      .map(e => Paths.get(e).toUri)
+    spark.client.artifactManager.addArtifacts(jars)
+  }
+
+  def createSparkSession(): SparkSession = {
+    if (!SystemUtils.isJavaVersionAtMost(JavaVersion.JAVA_17)) {
```

Comment:
This should be `if (!SystemUtils.isJavaVersionAtMost(JavaVersion.JAVA_16))` here, otherwise `SparkConnectServer` would not start up with Java 17 either ...

Reply:
This check is no longer needed, https://github.com/apache/spark/pull/42643 has been merged in, and to avoid conflicts in the connect module, https://github.com/apache/spark/pull/42643 has also been merged into branch-3.5.